#!/usr/bin/env python
PACKAGE = "mpc_node"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("max_speed",        double_t, 0, "max_speed",         1.5, 0.0, 100)
gen.add("path_length",      double_t, 0, "path_length",       6, 0.1, 100)
gen.add("goal_radius",      double_t, 0, "goal_radius",       0.2, 0.0, 1)
gen.add("controller_freq",  double_t, 0, "controller_freq",   10, 0.0, 100)
gen.add("mpc_steps",        double_t, 0, "mpc_steps",         20, 0.0, 100)
gen.add("mpc_ref_cte",      double_t, 0, "mpc_ref_cte",       0, 0.0, 1)
gen.add("mpc_ref_epsi",     double_t, 0, "mpc_ref_epsi",      0, 0.0, 1)
gen.add("mpc_ref_vel",      double_t, 0, "mpc_ref_vel",       1, 0.0, 10)
gen.add("mpc_w_cte",        double_t, 0, "mpc_w_cte",         1.0, 0.0, 1000)
gen.add("mpc_w_epsi",       double_t, 0, "mpc_w_epsi",        1.0, 0.0, 1000)
gen.add("mpc_w_vel",        double_t, 0, "mpc_w_vel",         0.5, 0.0, 1000)
gen.add("mpc_w_delta",      double_t, 0, "mpc_w_delta",       1.0, 0.0, 1000)
gen.add("mpc_w_accel",      double_t, 0, "mpc_w_accel",       2.0, 0.0, 1000)
gen.add("mpc_w_delta_d",    double_t, 0, "mpc_w_delta_d",     0.0, 0.0, 100)
gen.add("mpc_w_accel_d",    double_t, 0, "mpc_w_accel_d",     0.0, 0.1, 100)
gen.add("mpc_max_steering", double_t, 0, "mpc_max_steering",  0.436, 0.1, 10)
gen.add("mpc_max_throttle", double_t, 0, "mpc_max_throttle",  0.8, 0.0, 10)
gen.add("mpc_max_gama",     double_t, 0, "mpc_max_gama",      0.7, 0.0, 10)

exit(gen.generate(PACKAGE, "mpc_node", "mpc_dynamic"))

